app/models/user.rb

```diff
@@ -0,0 +1,43 @@
+class User < ApplicationRecord
+  attr_accessor :remember_token
+  
+  before_save { self.email = email.downcase }
+
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
+
+  validates :name, presence: true, length: { maximum: 50 } 
+  validates :email, presence: true, length: { maximum: 255 },
+                    format: { with: VALID_EMAIL_REGEX },
```

Comment:
Rails has a regex already in place for emails, you could it like: `format: { with: URI::MailTo::EMAIL_REGEXP }`